### PR TITLE
[docs] add diagrams and fix broken links

### PR DESCRIPTION
**Pull request by zihehuang**
## Description 

Added some diagrams that explains the escrow concepts better and fixed broken code sample link

## Test Plan 

How did you test the new or updated feature?
Visual inspection

---
If your changes are not user-facing and do not break anything, you can skip the following section. Otherwise, please briefly describe what has changed under the Release Notes section.

### Type of Change (Check all that apply)

- [ ] protocol change
- [ ] user-visible impact
- [ ] breaking change for a client SDKs
- [ ] breaking change for FNs (FN binary must upgrade)
- [ ] breaking change for validators or node operators (must upgrade binaries)
- [ ] breaking change for on-chain data layout
- [ ] necessitate either a data wipe or data migration

### CODE REVIEW

File: docs/content/guides/developer/sui-101/shared-owned.mdx [hunk 1]
```diff
@@ -37,11 +37,21 @@ The important property that this interface provides is that locked values cannot
...
-[Code Sample](https://github.com/MystenLabs/sui/blob/main/examples/move/escrow/sources/owned.move)
+[Code Sample](https://github.com/MystenLabs/sui/blob/93e6b4845a481300ed4a56ab4ac61c5ccb6aa008/examples/move/escrow/sources/owned.move)
 
-The protocol for swapping via escrow implemented using owned objects starts with both parties locking their respective objects. This will be used to prove that the object has not been tampered with after the swap has been agreed. If either party does not want to proceed at this stage, they just unlock their object.
+The protocol for swapping via escrow implemented using owned objects starts with both parties locking their respective objects. 
 
-Assuming both parties are happy to continue, the next step requires a third party to act as custodian.  The custodian holds objects that are waiting for their counterparts to arrive and when they arrive, it matches them up to complete the swap.
+    ![Escrow Step 1](../images/escrow-step1.png)
+
+This will be used to prove that the object has not been tampered with after the swap has been agreed. If either party does not want to proceed at this stage, they just unlock their object.
```
Comment (ronny-mysten) [inline docs/content/guides/developer/sui-101/shared-owned.mdx:46]:
```suggestion
This is used to prove that the object has not been tampered with after the swap has been agreed to. If either party doesn't want to proceed at this stage, they just unlock their object.
```

File: docs/content/guides/developer/sui-101/shared-owned.mdx [hunk 2]
```diff
@@ -37,11 +37,21 @@ The important property that this interface provides is that locked values cannot
...
+
+Assuming both parties are happy to continue, the next step requires both parties to swap the keys.
+
+    ![Escrow Step 2](../images/escrow-step2.png)
+
+We then introduce a third party to act as custodian. The custodian holds objects that are waiting for their counterparts to arrive and when they arrive, it matches them up to complete the swap.
```
Comment (ronny-mysten) [inline docs/content/guides/developer/sui-101/shared-owned.mdx:52]:
```suggestion
A third party acts as custodian. The custodian holds objects that are waiting for their counterparts to arrive and when they arrive, it matches them up to complete the swap.
```

**Review by ronny-mysten — COMMENTED** [review]
Can the images be made using [mermaid](https://mermaid.js.org/)? This allows for easier maintenance.

**Comment by zihehuang** [comment]
I will look into drawing the diagram with mermaid

**Review by ronny-mysten — APPROVED** [review]
Thanks for the mermaids. Makes maintenance a lot easier. The example used here has been changed, so made a Jira to look at this topic in the near future.

**Pull request merged**